added python codec templetes

buyalsky opened the pull request:
added extra condition for python folder naming convention

buyalsky committed:
added python codec templetes

buyalsky committed:
added python codec templetes

puzpuzpuz committed:
Revert protocol changes in MC#getCPMembers (#302)

* Revert protocol changes in MC#getCPMembers

* Migrate to UUID pairs in MC#getCPMembers

ihsandemir committed:
Removed back the addDistributedObjectListener isInternal field. (#305)

taburet committed:
Support bitmap indexes (#304)

* Support bitmap indexes

* fix ReferenceObjects

* make options nullable

mdumandag committed:
Add codec versioning support (#309)

* add ability to add new parameters to codecs

* update implementation

* Removed the hasNext check from the custom codec tempalate
since custom codecs always have an end frame at the end.
We should only check whether or not the next frame
is an end frame or not.

* Put newly added fix sized params to begin frame

We don't encode an initial frame if the custom type does not
have any fix sized parameters. However, if the custom type
is extended with a new fix sized parameter, new codecs
will try to encode/decode an initial frame which will
broke the communication between old clients/members
since they don't expect an initial frame. Therefore,
if the custom type is extended with a fix sized parameter,
we encode the data into the begin frame

* relax schma validation for the since values and check ordering of parameters

devOpsHazelcast commented:
[![CLA assistant check](http://ec2-54-211-157-189.compute-1.amazonaws.com:5000/pull/badge/not_signed)](http://ec2-54-211-157-189.compute-1.amazonaws.com:5000/hazelcast/hazelcast-client-protocol?pullRequest=303) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](http://ec2-54-211-157-189.compute-1.amazonaws.com:5000/hazelcast/hazelcast-client-protocol?pullRequest=303) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://ec2-54-211-157-189.compute-1.amazonaws.com:5000/check/hazelcast/hazelcast-client-protocol?pullRequest=303) it.</sub>

pertsodian committed:
Add triggerMapLoader param for putAll (#312)

abdulazizali77 committed:
Add MultiMap.putAll interface (#9079) (#313)

Introduce new EntryList_Data_List_Data type

Commit:
Supporting MemberInfo#addressMap (#316)

* supporting Memberinfo#addressMap

* changing "since" attr of newly added members from 2.0 to 2.1

* fixing template : missing whitespace after if (causes checkstyle errors in the generated code)

* representing ProtocolType with its ordinal

* changing to ProtocolType.WAN in reference_objects

Commit:
Correct since version as 2.1 for multimap.putall

Related wrongly merged pr.
https://github.com/hazelcast/hazelcast-client-protocol/pull/313

buyalsky committed:
add to_object keyword

buyalsky committed:
add to_object keyword

asimarslan committed:
ProtocolType enum is removed from protocol custom types (#324)

* ProtocolType enum is removed from protocol custom types and handled as builtin enum handling (as int)

* EnpointQualifier default value added

buyalsky committed:
add BitmapIndexOptionsCodec for python generator

devOpsHazelcast commented:
PR closed by Hazelcast automation as no activity (>3 months). Please reopen with comments, if necessary. Thank you for using Hazelcast and your valuable contributions
